### DISCUSSION TITLE
Where can I find something like ltsp-remoteapps ?

### DISCUSSION THREAD

**Discussion by abixadamj** [ltsp/ltsp#238] [Q&A]
I have new LTSP server. Previously uses could change their passwords via 
`lts-remoteapps passwd`
How I can do it similar way in new LTSP?

**Comment by alkisg** [comment]
The proper way would be for the users to be able to run `passwd` on the client, or any kind of password updating tool, and have a PAM module SSH to the server and update the password there instead. Similar to how PAM logins now work, completely transparently to the user and to applications.

This should be doable, patches most welcome.

LTSP remoteapps is a different thing; these too could be implemented in a non-LTSP specific way. They're basically a wrapper to `ssh -X`, they won't run on Wayland, and they have increasing issues from things like systemd-logind, dbus-daemon, remote notifications etc. They could also be implemented with "seamless RDP" in the future for Wayland compatibility.

Which part do you need, remote apps or password changing?

**Comment by abixadamj** [comment]
I need to have passwd for users. But I dont have knowledge how to do it.

**Comment by enaut** [comment]
I did write a graphical tool for users to be able to change their password. It's a small python script that calls ssh to the server using the old password and then calls passwd on the server.

It is packaged for 16.04 and the package runs on the newer verseions too. But I haven't  had time to package it for anything newer. It should be possible to just download the package (last link) and install it.

### Source:
https://github.com/enaut/password-changer
### PPA:
https://launchpad.net/~franz-die/+archive/ubuntu/ltsp-manager/
### Package:
https://launchpad.net/~franz-die/+archive/ubuntu/ltsp-manager/+files/python3-password-changer_1.0.0-ubuntu4_all.deb

**Comment by abixadamj** [comment]
I just did script using ssh - it works too ;-) But @enaut - I will remember to test your script later. Thank you all for helping. Tomorrow I will install this server in one scvhool in Warsaw (Poland). It will be my 25th servwer in Poland, but first with new LTSP.
